modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/MinimalApplicationModelGenerator.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.spring.dsl.model;
+
+import static org.mule.runtime.core.config.i18n.I18nMessageFactory.createStaticMessage;
+import org.mule.runtime.config.spring.dsl.api.ComponentBuildingDefinition;
+import org.mule.runtime.config.spring.dsl.processor.AbstractAttributeDefinitionVisitor;
+import org.mule.runtime.core.api.MuleRuntimeException;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Generates the minimal required component set to create an isolated configuration
+ * component (i.e.: file:config, ftp:connection, a flow MP).
+ *
+ * Based on the requested component, the {@link ComponentModel} configuration associated is
+ * introspected to find it dependencies based on it's {@link ComponentBuildingDefinition}.
+ * This process is recursively done for each of the dependencies in order to find all the required
+ * {@link ComponentModel}s that must be created for the requested {@link ComponentModel} to
+ * work properly.
+ *
+ * @since 4.0
+ */
+public class MinimalApplicationModelGenerator {
+
+  private final ComponentBuildingDefinitionRegistry componentBuildingDefinitionRegistry;
+  private ApplicationModel applicationModel;
+
+  /**
+   * Creates a new instance associated to a complete {@link ApplicationModel}.
+   *
+   * @param applicationModel the artifact {@link ApplicationModel}.
+   * @param componentBuildingDefinitionRegistry the registry to find the {@link ComponentBuildingDefinition}s  associated to each {@link ComponentModel} that must be resolved.
+     */
+  public MinimalApplicationModelGenerator(ApplicationModel applicationModel,
+                                          ComponentBuildingDefinitionRegistry componentBuildingDefinitionRegistry) {
+    this.applicationModel = applicationModel;
+    this.componentBuildingDefinitionRegistry = componentBuildingDefinitionRegistry;
+  }
+
+  /**
+   * Resolves the minimal set of {@link ComponentModel}s for a message processor
+   * within a flow.
+   *
+   * @param processorPath the processor path in which the message processor is located.
+   * @return the generated {@link ApplicationModel} with the minimal set of {@link ComponentModel}s required.
+   * @throws NoSuchComponentModelException if the requested component does not exists.
+   */
+  public ApplicationModel getMinimalModelByPath(String processorPath) {
```

Comment:
For now yes, but we will add probably references to sources in the future. I will rename this to componentPath.